rusk-wallet/src/bin/main.rs

```diff
@@ -158,6 +158,24 @@ async fn exec() -> anyhow::Result<()> {
         }
     };
 
+    if !settings.check_state_con().await? {
+        error!(
+            "Error while checking the state endpoint status {}",
+            settings.state,
+        );
+
+        return Ok(());
+    }
+
+    if !settings.check_prover_con().await? {
+        error!(
+            "Error while checking the prover endpoint status {}",
+            settings.prover,
+        );
+
+        return Ok(());
+    }
+
```

Comment:
We cannot exit if there is no connection!

User should still be able to do some operation, that's why there was a different menu for when the client were offline (now that menu is displayed when wallet is not synced, idk why)